src/radical/pilot/pmgr/launching/psi_j.py

```diff
@@ -0,0 +1,186 @@
+
+        schema = schemas[0]
+
+        if schema == 'fork':
+            schema = 'local'
+
+        return schema
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _translate_state(self, status):
+
+        if   status.state == psij.JobState.NEW       : return rps.NEW
+        elif status.state == psij.JobState.QUEUED    : return rps.PMGR_LAUNCHING
+        elif status.state == psij.JobState.ACTIVE    : return rps.PMGR_ACTIVE
+        elif status.state == psij.JobState.COMPLETED : return rps.DONE
+        elif status.state == psij.JobState.FAILED    : return rps.FAILED
+        elif status.state == psij.JobState.CANCELED  : return rps.CANCELED
+        else:
+            raise ValueError('cannot interpret psij state: %s' % repr(status))
+
+    # --------------------------------------------------------------------------
+    #
+    def _job_status_cb(self, job, status):
+
+        self._called = True
+
+        try:
+            with self._lock:
+
+                if job.id not in self._pilots:
+                    return
+
+                rp_state = self._translate_state(status)
+                pilot    = self._pilots[job.id]
+
+            self._state_cb(pilot, rp_state)
+
+        except Exception:
+            self._log.exception('job status callback failed')
+
+
+    # --------------------------------------------------------------------------
+    #
+    def can_launch(self, rcfg, pilot):
+
+        schema = self._get_schema(rcfg)
+
+        if not schema:
+            return False
+
+        if schema not in self._jex:
+
+            self._log.debug('=== create executor for %s', schema)
+            try:
+                self._jex[schema] = psij.JobExecutor.get_instance(schema)
+                self._jex[schema].set_job_status_callback(self._job_status_cb)
+            except:
+                self._log.exception('failed to create psij executor')
+                return False
+
+        return True
+
+
+    # --------------------------------------------------------------------------
+    #
+    def launch_pilots(self, rcfg, pilots):
+
+        assert(psij)
+
+        for pilot in pilots:
+
+            pid    = pilot['uid']
+            schema = self._get_schema(rcfg)
+            assert(schema)
+
+            jex = self._jex.get(schema)
+
+            assert(jex)
+
+            jd   = pilot['jd_dict']
+            spec = psij.JobSpec()
+
+            spec.executable  = jd['executable']
+            spec.arguments   = jd['arguments']
+            spec.environment = jd['environment']
+            spec.directory   = jd['working_directory']
+            spec.stdout_path = jd['output']
+            spec.stderr_path = jd['error']
+
+            spec.resources   = psij.ResourceSpecV1()
+            spec.resources.process_count         = 1
+            spec.resources.cpu_cores_per_process = jd['total_cpu_count']
+            spec.resources.gpu_cores_per_process = jd['total_gpu_count']
```

Comment:
`process_count = jd.total_cpu_count`? (since `total_cpu_count = (requested_nodes * cores_per_node) or requested_cores`)
and `processes_per_node = jd.processes_per_host`?

Reply:
Semantics is somewhat different here than in SAGA.  Basically, `process_count = 2` would mean 2 ranks, so we would get 2 bootstrappers.

Reply:
I also used a corresponding template to check how it is mapped - https://github.com/ExaWorks/psi-j-python/blob/main/src/psij/executors/batch/slurm/slurm.mustache

Reply:
Not sure what you try to say, sorry :-)

Reply:
TODO AM: fix

Reply:
Fixed this now, but I am still uncertain about it really because we had to take out the GPU spec.  Assume a scheduler is clever enough to distinguish between GPU and CPU nodes - we could now suddenly scheduled on CPU-only nodes because we don't include GPU requests.

Anyway, that is more an issue for psij really than for RP to solve.  Let's see how it goes and if it breaks we should address it in psij.

Reply:
btw, right away we should add `project` and `queue` to psij job (usually cpu-only and gpu nodes are provided in different queues)

Reply:
Right, I added `project`, `queue` and the other job attributes.

`process_per_node` doesn't really fit (unless it means something different than the name suggests), as we want a single process over the whole allocation (several nodes).  I have a deja-vu so, I think we discussed this already?  Let me check again...

Reply:
this is part of `spec.resources` - https://github.com/ExaWorks/psi-j-python/blob/main/src/psij/resource_spec.py#L50